### PR TITLE
fix(general): Delete duplicates in time of day array within policies

### LINKED ISSUE #3441: UI problem with version 0.15.0 (time of day)
Hello. I would like to bring up another interface problem:
When you define a time of day, in effective you see it 3 times. Have a nice day!

### PR DESCRIPTION
**Pull request by lupusA**
Hi, 

this PR fixes #3441. The issue was the the function ```SortAndDedupeTimesOfDay``` within ```scheduling_policy.go``` only sorted the slice. Duplicates were not deleted. The PR uses the ```slices.SortFunc``` function to first sort the slice. 

After that the sorted slice is compacted to remove subsequent duplicate entries. 

Feedback is welcomed. 

Cheers,

### CODE REVIEW

File: snapshot/policy/scheduling_policy.go [hunk 1]
```diff
@@ -45,14 +46,18 @@ func (t TimeOfDay) String() string {
 
 // SortAndDedupeTimesOfDay sorts the slice of times of day and removes duplicates.
 func SortAndDedupeTimesOfDay(tod []TimeOfDay) []TimeOfDay {
-	sort.Slice(tod, func(i, j int) bool {
-		if a, b := tod[i].Hour, tod[j].Hour; a != b {
-			return a < b
+	slices.SortFunc(tod, func(a, b TimeOfDay) int {
```
Comment (julio-lopez) [inline snapshot/policy/scheduling_policy.go:49]: This is great 🎉 

@lupusA Can you add a simple ("table-based") unit test for `SortAndDedupeTimesOfDay` please?

**Review by julio-lopez — COMMENTED** [review]
This looks great.

Can you add a unit test please?

Thanks

File: snapshot/policy/scheduling_policy.go [hunk 2]
```diff
@@ -45,14 +46,18 @@ func (t TimeOfDay) String() string {
...
+		if n := cmp.Compare(a.Hour, b.Hour); n != 0 {
+
```
Comment (julio-lopez) [inline snapshot/policy/scheduling_policy.go:51]: nit: the linter is complaining about this extra line.

```suggestion
```

I used `gofumpt -d  snapshot/policy/scheduling_policy.go` to find out.

```diff
diff snapshot/policy/scheduling_policy.go.orig snapshot/policy/scheduling_policy.go
--- snapshot/policy/scheduling_policy.go.orig
+++ snapshot/policy/scheduling_policy.go
@@ -48,7 +48,6 @@
 func SortAndDedupeTimesOfDay(tod []TimeOfDay) []TimeOfDay {
 	slices.SortFunc(tod, func(a, b TimeOfDay) int {
 		if n := cmp.Compare(a.Hour, b.Hour); n != 0 {
-
 			return n
 		}
```

File: snapshot/policy/scheduling_policy.go (hunk 1 above)
Comment (lupusA) [reply snapshot/policy/scheduling_policy.go:49]: Hey @julio-lopez 
will provide the test in the next days

File: snapshot/policy/scheduling_policy.go (hunk 2 above)
Comment (lupusA) [reply snapshot/policy/scheduling_policy.go]: Fixed with latest commit

File: snapshot/policy/scheduling_policy.go (hunk 1 above)
Comment (julio-lopez) [reply snapshot/policy/scheduling_policy.go:49]: Test in #3535
The test passes with this changes.

**Review by julio-lopez — APPROVED** [review]

**Pull request merged**